compliance_checker/cf/cf.py

```diff
@@ -1893,14 +1894,12 @@ def check_calendar(self, ds):
                 continue
             reasoning = None
             has_calendar = hasattr(v, 'calendar')
-            if not has_calendar:
-                reasoning = ['Variable %s should have a calendar attribute' % k]
             result = Result(BaseCheck.LOW,
                             has_calendar,
                             '§4.4.1 Time and calendar',
                             reasoning)
-            ret_val.append(result)
-
+            # if has a calendar, check that it is within the valid values
+            # otherwise no calendar is valid
```

Comment:
this can be interpreted two ways (thanks, english) but not pressing enough to correct this PR